include/stream-client/detail/logger.ipp

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+namespace stream_client {
+
+namespace logging {
+
+inline const log_interface* get_logger(const log_interface* new_logger = nullptr) {
+    static const log_interface* glob = nullptr;
+    if (new_logger) {
+        glob = new_logger;
+    }
+    return glob;
+}
+
+inline void LOG_ERROR(const std::string& target, const std::string& message, const std::runtime_error& error) {
+    const log_interface* logger = get_logger();
+    if (logger) {
+        logger->message(log_level::error, target, message, &error);
+    }
+}
+
+inline void LOG_WARN(const std::string& target, const std::string& message) {
+    const log_interface* logger = get_logger();
+    if (logger) {
+        logger->message(log_level::warning, target, message);
+    }
+}
+
+inline void LOG_INFO(const std::string& target, const std::string& message) {
+    const log_interface* logger = get_logger();
+    if (logger) {
+        logger->message(log_level::info, target, message);
+    }
+}
+
+class std_streams_logger: public log_interface
```

Comment:
This class should be in the `detail` namespace as well since it is purely internal and not supposed to be used by the user